src/appleseed.bench/mainwindow/applicationsettingswindow.cpp

```diff
@@ -0,0 +1,151 @@
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "applicationsettingswindow.h"
+
+// UI definition header.
+#include "ui_applicationsettingswindow.h"
+
+// appleseed.bench headers.
+#include "utility/settingskeys.h"
+
+// appleseed.qtcommon headers.
+#include "utility/miscellaneous.h"
+
+// appleseed.renderer headers.
+#include "renderer/api/utility.h"
+
+// appleseed.foundation headers.
+#include "foundation/platform/system.h"
+#include "foundation/utility/log/logmessage.h"
+#include "foundation/utility/string.h"
+
+// Qt headers.
+#include <QKeySequence>
+#include <QShortcut>
+#include <Qt>
+
+// Standard headers.
+#include <string>
+
+using namespace appleseed::qtcommon;
+using namespace foundation;
+using namespace renderer;
+
+namespace appleseed {
+namespace bench {
+
+//
+// ApplicationSettingsWindow class implementation.
+//
+
+ApplicationSettingsWindow::ApplicationSettingsWindow(ParamArray& settings, QWidget* parent)
+  : WindowBase(parent, "application_settings_window")
+  , m_ui(new Ui::ApplicationSettingsWindow())
+  , m_settings(settings)
+{
+    m_ui->setupUi(this);
+    setWindowFlags(Qt::Window);
+    build_connections();
+    WindowBase::load_settings();
+    load_settings();
+}
+
+ApplicationSettingsWindow::~ApplicationSettingsWindow()
+{
+    delete m_ui;
+}
+
+void ApplicationSettingsWindow::slot_reload_application_settings()
+{
+    load_settings();
+}
+
+void ApplicationSettingsWindow::build_connections()
+{
+    connect(m_ui->buttonbox, SIGNAL(accepted()), SLOT(slot_save_configuration_and_close()));
+    connect(m_ui->buttonbox, SIGNAL(rejected()), SLOT(slot_restore_configuration_and_close()));
+
+    connect(
+        create_window_local_shortcut(this, Qt::Key_Return), SIGNAL(activated()),
+        SLOT(slot_save_configuration_and_close()));
+
+    connect(
+        create_window_local_shortcut(this, Qt::Key_Enter), SIGNAL(activated()),
+        SLOT(slot_save_configuration_and_close()));
+
+    connect(
+        create_window_local_shortcut(this, Qt::Key_Escape), SIGNAL(activated()),
+        SLOT(slot_restore_configuration_and_close()));
+}
+
+void ApplicationSettingsWindow::load_settings()
+{
+    // Exploit the fact that category values match combobox item indices exactly.
+    const std::string message_verbosity =
+        m_settings.get_path_optional<std::string>(SETTINGS_MESSAGE_VERBOSITY, "info");
+    m_ui->combobox_message_verbosity->setCurrentIndex(
+        message_verbosity == "debug"   ? 0 :
+        message_verbosity == "info"    ? 1 :
+        message_verbosity == "warning" ? 2 :
+        message_verbosity == "error"   ? 3 :
+        message_verbosity == "fatal"   ? 4 :
+        1);     // info if an unknown value was found
+}
+
+void ApplicationSettingsWindow::save_settings()
+{
+    // Exploit the fact that category values match combobox item indices exactly.
+    const auto sampling_mode_index = m_ui->combobox_message_verbosity->currentIndex();
```

Comment:
`sampling_mode_index` ? Don't you mean `message_verbosity ` ?

Reply:
Good catch!